incubator/nodejs-loopback/README.md

```diff
@@ -0,0 +1,30 @@
+# Node.js LoopBack Stack
+
+The Node.js LoopBack stack extends the [Node.js stack](../nodejs/README.md) and provides a powerful solution
+to build open APIs ans micro-services in TypeScript with [LoopBack](https://loopback.io/), an open source Node.js API framework. It is based on [LoopBack 4](https://github.com/strongloop/loopback-next).
+
+## Templates
+
+- `scaffold`: The template is generated with `lb4` command from `@loopback/cli`. See https://loopback.io/doc/en/lb4/Getting-started.html for more information.
```

Comment:
@bajtos I think you have raised a good question. I was wondering about the versioning support for stacks too. A few thoughts:

1. The quick and dirty way is to create a new stack for a new version and name it as `nodejs-loopback-v1.0.1`

2. It would be ideal to allow stacks to be published to `npm` or `maven`, which will have versioning capabilities for free. Then they can be registered or discovered.

Reply:
@raymondfeng @bajtos
Appsody stacks are versioned and the versioning is managed by `stack.yaml` file. The build system will use this to build new docker images and make newer temapltes available.

Reply:
@neeraj-laad I understand the images are versioned. But can we have multiple versions of the template in the github repo? Or does it make sense?

Reply:
My main concern is about DX for people maintaining this template. When we introduce changes in LB4, for example we call a new sequence action from the scaffolded sequence class, or change the way how monitoring & metrics endpoints are added, how are we going to update the appsody template to match the output of the latest `lb4` CLI tool?

More specifically:
- What are the steps to determine what has changed in the output of `lb4 app` and which part of these changes to bring into appsody template?
- What process will we use to ensure the appsody template is updated after we make changes in `lb4` CLI?

I am concerned that the current proposal will make it very easy to forget to bring the appsody template up-to-date frequently enough, and also make it easy to accidentally re-introduce files like Dockerfile that we decided to remove as part of this initial pull request.

The second concern can be solved by automated test suite here in appsody that will verify that new version are not bringing back artifacts that don't belong to the template.

I would like to propose two alternative approaches:

(1)
Keep the LB template inside loopback-next monorepo, so that we can make changes both to `lb4` CLI and appsody template as part of a single pull request. Add a new item to our PR checklist to remind us to update the appsody template (see https://raw.githubusercontent.com/strongloop/loopback-next/master/.github/PULL_REQUEST_TEMPLATE.md)

(2)
Don't lock the LB template inside appsody, modify the project generator to call `lb4` CLI to create project files. We can introduce a new flag to `lb4 app` command to tell it to create appsody-based project.